src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAuthenticationManager.java

```diff
@@ -498,24 +499,41 @@ public RegisteredUser createIncompleteLoginUserSession(final HttpServletRequest
 
     /**
      * Destroy a session attached to the request.
-     * 
-     * @param request
-     *            containing the tomcat session to destroy
-     * @param response
-     *            to destroy the segue cookie.
+     *
+     * @param request  containing the tomcat session to destroy
+     * @param response to destroy the segue cookie.
+     * @throws NoUserLoggedInException - if a user cannot be retrieved from the session information
+     * @throws SegueDatabaseException  - if accessing the database fails
      */
-    public void destroyUserSession(final HttpServletRequest request, final HttpServletResponse response) {
+    public void destroyUserSession(final HttpServletRequest request, final HttpServletResponse response) throws NoUserLoggedInException, SegueDatabaseException  {
         Validate.notNull(request);
         try {
             request.getSession().invalidate();
+            invalidateSessionToken(request);
             Cookie logoutCookie = createAuthLogoutCookie();
 
-            response.addCookie(logoutCookie);  // lgtm [java/insecure-cookie]  false positive due to conditional above!
+            response.addCookie(logoutCookie);
         } catch (IllegalStateException e) {
-            log.info("The session has already been invalidated. " + "Unable to logout again...", e);
+            log.info(LOGOUT_SESSION_ARLEADY_INVALIDATED_MESSAGE, e);
         }
     }
-    
+
+    /**
+     * Takes a request holding an authentication cookie and invalidates the associated session token stored in the database
+     *
+     * @param request - a servlet request holding an auth cookie for the user session to be invalidated
+     * @throws NoUserLoggedInException - if a user cannot be retrieved from the session information
+     * @throws SegueDatabaseException  - if accessing the database fails
+     */
+    public void invalidateSessionToken(final HttpServletRequest request) throws NoUserLoggedInException, SegueDatabaseException {
+        Validate.notNull(request);
+        RegisteredUser currentUser = this.getUserFromSession(request, false);
+        if (null == currentUser) {
+            throw new NoUserLoggedInException();
+        }
```

Comment:
`//by regenerating the token without updating it in the cookie we stop them matching and invalidate the session`